File: tests/fixtures/language_pack/src/cpp/widget.cpp
#include "widget.hpp"

namespace languagepack {

const int CPP_LIMIT = 4;

class CppWidget {
public:
    void render() {
        const char *ignored = "class CppStringFake {}";
    }
};

struct CppOptions {
    int limit;
};

enum class CppMode {
    Compact
};

int build_cpp_widget(
    int value
) {
    return value;
}

}

File: tests/fixtures/language_pack/src/csharp/Widget.cs
using System.Collections.Generic;

namespace LanguagePack.CSharp
{
    public interface CSharpRenderable
    {
        void Render();
    }

    public struct CSharpOptions
    {
        public int Limit;
    }

    public enum CSharpMode
    {
        Compact
    }

    public record CSharpRecord(string Name);

    public class CSharpWidget
    {
        private const int CSharpLimit = 4;

        public string Name { get; }

        public CSharpWidget(string name)
        {
            Name = name;
        }

        public void Render(
            string message
        )
        {
            var ignored = "class CSharpStringFake {}";
        }
    }
}
